openmc/stats/univariate.py

```diff
@@ -0,0 +1,827 @@
+            return Maxwell.from_xml_element(elem)
+        elif distribution == 'watt':
+            return Watt.from_xml_element(elem)
+        elif distribution == 'normal':
+            return Normal.from_xml_element(elem)
+        elif distribution == 'muir':
+            return Muir.from_xml_element(elem)
+        elif distribution == 'tabular':
+            return Tabular.from_xml_element(elem)
+        elif distribution == 'legendre':
+            return Legendre.from_xml_element(elem)
+        elif distribution == 'mixture':
+            return Mixture.from_xml_element(elem)
+
+
+class Discrete(Univariate):
+    """Distribution characterized by a probability mass function.
+
+    The Discrete distribution assigns probability values to discrete values of a
+    random variable, rather than expressing the distribution as a continuous
+    random variable.
+
+    Parameters
+    ----------
+    x : Iterable of float
+        Values of the random variable
+    p : Iterable of float
+        Discrete probability for each value
+
+    Attributes
+    ----------
+    x : Iterable of float
+        Values of the random variable
+    p : Iterable of float
+        Discrete probability for each value
+
+    """
+
+    def __init__(self, x, p):
+        super().__init__()
+        self.x = x
+        self.p = p
+
+    def __len__(self):
+        return len(self.x)
+
+    @property
+    def x(self):
+        return self._x
+
+    @property
+    def p(self):
+        return self._p
+
+    @x.setter
+    def x(self, x):
+        if isinstance(x, Real):
+            x = [x]
+        cv.check_type('discrete values', x, Iterable, Real)
+        self._x = x
+
+    @p.setter
+    def p(self, p):
+        if isinstance(p, Real):
+            p = [p]
+        cv.check_type('discrete probabilities', p, Iterable, Real)
+        for pk in p:
+            cv.check_greater_than('discrete probability', pk, 0.0, True)
+        self._p = p
+
+    def to_xml_element(self, element_name):
+        """Return XML representation of the discrete distribution
+
+        Parameters
+        ----------
+        element_name : str
+            XML element name
+
+        Returns
+        -------
+        element : xml.etree.ElementTree.Element
+            XML element containing discrete distribution data
+
+        """
+        element = ET.Element(element_name)
+        element.set("type", "discrete")
+
+        params = ET.SubElement(element, "parameters")
+        params.text = ' '.join(map(str, self.x)) + ' ' + ' '.join(map(str, self.p))
+
+        return element
+
+    @classmethod
+    def from_xml_element(cls, elem):
+        """Generate discrete distribution from an XML element
+
+        Parameters
+        ----------
+        elem : xml.etree.ElementTree.Element
+            XML element
+
+        Returns
+        -------
+        openmc.stats.Discrete
+            Discrete distribution generated from XML element
+
+        """
```

Comment:
Same suggestion in this file as I had in `multivariate.py` for the `from_xml_element` methods. They'd benefit from Error handling for bad xmls.